website/docs/guides/version_4_upgrade.html.markdown

```diff
@@ -0,0 +1,295 @@
+update the version constraints in your Terraform configuration and run
+[`terraform init`](https://www.terraform.io/docs/commands/init.html) to download
+the new version.
+
+If you aren't using version constraints, you can use `terraform init -upgrade`
+in order to upgrade your provider to the latest released version.
+
+For example, given this previous configuration:
+
+```hcl
+provider "vault" {
+  # ... other configuration ...
+
+  version = "~> 3.25.0"
+}
+```
+
+An updated configuration:
+
+```hcl
+provider "vault" {
+  # ... other configuration ...
+
+  version = "~> 4.0.0"
+}
+```
+
+## Provider Policy Changes
+
+Version `4.0.0` of the Vault provider made changes to the underlying Vault API
+calls, which in turn may require policy adjustments in environments where
+permissions are least privilege.
+
+Please see the [Capabilities](https://developer.hashicorp.com/vault/docs/concepts/policies#capabilities)
+section of the Vault Policies documentation for more information on Vault
+policies.
+
+### Auth method resource changes
+
+The below table specifies what changed between version 3.X and 4.X for the
+following resources:
+
+#### Data sources
+- `vault_auth_backend`
+
+#### Resources
+- `vault_auth_backend`
+- `vault_gcp_auth_backend`
+- `vault_github_auth_backend`
+- `vault_jwt_auth_backend`
+- `vault_ldap_auth_backend`
+- `vault_okta_auth_backend`
+
+<table>
+<thead>
+  <tr>
+    <th colspan="2">3.X</th>
+    <th colspan="2">4.X</th>
+  </tr>
+</thead>
+<tbody>
+  <tr>
+    <td>Method</td>
+    <td>Path</td>
+    <td>Method</td>
+    <td>Path</td>
+  </tr>
+  <tr>
+    <td>GET</td>
+    <td>sys/auth</td>
+    <td>GET</td>
+    <td>sys/auth/:path</td>
+  </tr>
+</tbody>
+</table>
+
+For example, in version 3.X the following policy is required for the
+`vault_gcp_auth_backend` resource to perform the GET `sys/auth` HTTP operation:
+
+```hcl
+path "sys/auth"
+{
+    capabilities = ["read"]
+}
+```
+
+In version 4.X the following policy is required for the
+`vault_gcp_auth_backend` resource to perform the GET `sys/auth/:path` HTTP operation:
+
+```hcl
+path "sys/auth/gcp"
+{
+    capabilities = ["read"]
+}
+```
+
+### Secret engine resource changes
+
+The below table specifies what changed between version 3.X and 4.X for the
+following resources:
+
+#### Resources
+  - `vault_aws_secret_backend`
+  - `vault_azure_secret_backend`
+  - `vault_consul_secret_backend`
+  - `vault_gcp_secret_backend`
+  - `vault_mount`
+  - `vault_pki_secret_backend_cert`
+  - `vault_rabbitmq_secret_backend`
+  - `vault_terraform_cloud_secret_backend`
+
+<table>
+<thead>
+  <tr>
+    <th colspan="2">3.X</th>
+    <th colspan="2">4.X</th>
+  </tr>
+</thead>
+<tbody>
+  <tr>
+    <td>Method</td>
+    <td>Path</td>
+    <td>Method</td>
+    <td>Path</td>
+  </tr>
+  <tr>
+    <td>GET</td>
+    <td>sys/mounts</td>
+    <td>GET</td>
+    <td>sys/mounts/:path</td>
+  </tr>
+</tbody>
+</table>
+
+For example, in version 3.X the following policy is required for the
```

Comment:
Yes, I would guess most users are using a highly privileged token. The assumptions I made when writing this guide were 
- the Upgrade Guide should be as concise as possible
- users running TFVP with tightly controlled token permissions already have an understanding of Vault Policies and TFVP
- they do not need guidance on TFVP's existing policy requirements
- they would be able to read the guide and see what they need to change

It may be that this view is too narrow and we should also cater to a wider audience, i.e. users that aren't using the principle of least privilege when running TFVP. I am open to suggestions here. :)